Add eslint rule to force devs to memoize values sent to context providers

rodrigofariow opened the pull request:
Relates to https://github.com/visgl/react-google-maps/pull/287

Also added a very basic, not so opinionated settings.json and extensions.json file with prettier correctly configured.
Sounds like a good thing to have on the git repo without causing issues

rodrigofariow commented on .eslintrc.js:
```diff
@@ -44,6 +44,7 @@ module.exports = {
         'no-undef': 'off',
         'react-hooks/rules-of-hooks': 'error',
         'react-hooks/exhaustive-deps': 'warn',
```
Side note (non blocking):
I would also set this rule to "error" since this can be critical, specially for a library like this one.

usefulthink replied:
Good point – lets do that as well...

I wish there would be a bit more fine-grained control over this, we only have the option to disable the entire rule where we need to. Would be awesome if we could tell eslint to just ignore one specific variable used in a hook.

usefulthink reviewed (approved):
Thanks a lot! I wasn't even aware of that rule.

But please remove the vscode config files, which editor and editor-configurations to use is a personal decision we don't want to interfere with. 

Also that would likely lead to endless accidental commits of peoples vscode configurations.

rodrigofariow commented:
> Thanks a lot! I wasn't even aware of that rule.

Yep, it's pretty awesome. We added it at my company's codebase and it has caught a lot of perf regressions. Should be more wildly used for sure!

@usefulthink 
Removed the vscode files and set the exhaustive deps rule to error.

Cheers
